examples/with-sui/src/index.ts

```diff
@@ -0,0 +1,164 @@
+import * as dotenv from "dotenv";
+import * as path from "path";
+import {
+  JsonRpcProvider,
+  Ed25519PublicKey,
+  TransactionBlock,
+  mainnetConnection,
+  messageWithIntent,
+  IntentScope,
+  toSerializedSignature,
+} from "@mysten/sui.js";
+import { input } from "@inquirer/prompts";
+import { Turnkey } from "@turnkey/sdk-server";
+import { blake2b } from "@noble/hashes/blake2b";
+import { bytesToHex } from "@noble/hashes/utils";
+
+dotenv.config({ path: path.resolve(process.cwd(), ".env.local") });
+
+async function main() {
+  const organizationId = process.env.ORGANIZATION_ID!;
+  const turnkeyClient = new Turnkey({
+    apiBaseUrl: process.env.BASE_URL!,
+    apiPublicKey: process.env.API_PUBLIC_KEY!,
+    apiPrivateKey: process.env.API_PRIVATE_KEY!,
+    defaultOrganizationId: organizationId,
+  });
+
+  const provider = new JsonRpcProvider(mainnetConnection);
+
+  // Use your Sui address and public key from environment variables
+  const suiAddress = process.env.SUI_ADDRESS!;
+  const suiPublicKeyHex = process.env.SUI_PUBLIC_KEY!;
+
+  if (!suiAddress || !suiPublicKeyHex) {
+    console.error(
+      "Please set your SUI_ADDRESS and SUI_PUBLIC_KEY in the .env.local file."
+    );
+    process.exit(1);
+  }
+
+  console.log(`Using Sui address: ${suiAddress}`);
+  const publicKeyBytes = Buffer.from(suiPublicKeyHex, "hex");
+
+  // Verify that the public key corresponds to the Sui address
+  const publicKey = new Ed25519PublicKey(publicKeyBytes);
+  const computedAddress = publicKey.toSuiAddress();
+
+  if (computedAddress !== suiAddress) {
+    console.error("The SUI_PUBLIC_KEY does not correspond to the SUI_ADDRESS.");
+    process.exit(1);
+  }
+
+  // Check balance
+  const balanceData = await provider.getBalance({
+    owner: suiAddress,
+    coinType: "0x2::sui::SUI",
+  });
+  const balance = BigInt(balanceData.totalBalance);
+  if (balance === 0n) {
+    console.log(
+      `Your balance is zero. Please fund your address ${suiAddress} to proceed.`
+    );
+    process.exit(1);
+  }
+
+  // Fetch the user's SUI coin objects
+  const coins = await provider.getCoins({
+    owner: suiAddress,
+    coinType: "0x2::sui::SUI",
+  });
+
+  if (coins.data.length === 0) {
+    console.error("No SUI coins found in the account.");
+    process.exit(1);
+  }
+
+  // Create and sign a transaction
+  const recipientAddress = await input({
+    message: "Recipient address:",
+    default: "<recipient_sui_address>",
```

Comment:
we could create a SUI warchest that we default to here

Reply:
It's mainnet so idk how I'd feel about that 😂

Reply:
LOLOL trueeee i take that back